### DISCUSSION TITLE
Is there an events list somewhere in the documentation?

### DISCUSSION THREAD

**Discussion by flavi1** [copy/v86#888] [Q&A] [answered]
Hello.
I'm looking for a list of events emited by the emulator object, but i can't find it on the documentation.
In the starter.js file, a comment is about a "events.md" page.
Does this page already exists?

If not, I just want to make a progress bar for the initial state file loading.
How to start to do it?

Thank you for this project!

**Comment by SuperMaxusa** [comment, marked as answer]
> If not, I just want to make a progress bar for the initial state file loading.
> How to start to do it?

most likely you need to use **"download_progress"** event for this as here:
https://github.com/copy/v86/blob/b81df778019dd1fd43f7c813aad7266d30f626cb/src/browser/main.js#L1429-L1432
https://github.com/copy/v86/blob/b81df778019dd1fd43f7c813aad7266d30f626cb/src/browser/main.js#L78-L92

**Reply by flavi1** [reply]
It works. And a lot of event listeners are defined in this file. 
Thank you.
